apis/appmesh/v1beta2/types.go

```diff
@@ -189,6 +189,16 @@ type Listener struct {
 	TLS *ListenerTLS `json:"tls,omitempty"`
 }
 
+// VirtualGatewayReference holds a reference to VirtualGateway.appmesh.k8s.aws
+type VirtualGatewayReference struct {
+	// Namespace is the namespace of VirtualGateway CR.
+	// If unspecified, defaults to the referencing object's namespace
+	// +optional
+	Namespace *string `json:"namespace,omitempty"`
+	// Name is the name of VirtualGateway CR
+	Name string `json:"name"`
```

Comment:
better to add a UID here for system populated reference, to make sure the real object referenced is same one.